dataactvalidator/filestreaming/csvReader.py

```diff
@@ -67,12 +83,22 @@ def get_writer(bucket_name, filename, header, is_local, region=None):
         """
         Gets the write type based on if its a local install or not.
         """
+        # TODO verify that the writer works corectly for local and S3
         if is_local:
             return CsvLocalWriter(filename, header)
         if region is None:
             region = CONFIG_BROKER["aws_region"]
         return CsvS3Writer(region, bucket_name, filename, header)
 
+    def _transfer_s3_file_to_local(self, bucket, filename):
+        (file, file_path) = tempfile.mkstemp()
+        file.close()
+
+        s3 = boto3.client('s3')
+        s3.download_file(bucket, filename, file_path)
```

Comment:
Downloading the file locally and reading from there could be a potential risk with storage and memory. If this is the fastest route to take, cleanup of the downloaded files will need to happen.

Also, will this be less memory intensive as the byte-streaming that was happening previously? If we're reading in extremely large files, it may have an adverse impact. It should be tested with a realistic file set.